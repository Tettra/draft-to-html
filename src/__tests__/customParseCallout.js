import { convertFromRaw } from 'draft-js'
import convertToHtml from '../'
import calloutBlock from './calloutBlock.json'

const parseBlock = (block, contentState) => {
  if (block.getType().toLowerCase() === 'callout') {
    const el = document.createElement('blockquote')
    return el
  }
}

describe('convertToHtml with custom parseBlock option', () => {
  test('converts calloutBlock with custom parseBlock function', () => {
    const html = convertToHtml(convertFromRaw(calloutBlock), { parseBlock })
    expect(html).toMatchSnapshot()
  })
})
