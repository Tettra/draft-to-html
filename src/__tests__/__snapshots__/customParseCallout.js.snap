// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertToHtml with custom parseBlock option converts calloutBlock with custom parseBlock function 1`] = `"<blockquote>Hello there frien<strong>d this</strong> is a callo<em>ut bloc</em>k yep</blockquote>"`;
